### ISSUE TITLE
Gatsby PoC documentation site

### ISSUE THREAD

**Issue by ty2k** [BCDevOps/developer-experience#2263]
Set up a [Gatsby](https://www.gatsbyjs.com/) instance for use as a documentation site with source Markdown documents stored in a mono-repo (TBD).

The intention with this proof of concept work is to demonstrate the typical pattern of documentation living alongside the code needed to build the application that publishes it, all in a single repository. In this model, documentation additions, changes, and deletions can be reflected in new builds of the application triggered by the pull requests that alter the state of the documentation.

This contrasts with the disparate repository model we currently use for our [documentation pages in DevHub](https://developer.gov.bc.ca/). In the DevHub model, community documents can be hosted in their own repositories (including under different organizations). A new build of the DevHub application is required to pull in the latest changes from all repositories, which requires either a cron job or manual action from a platform services team member.

**Additional context**
Use the `bdec76-dev` namespace.

See [#1412](https://github.com/BCDevOps/developer-experience/issues/1412) for extra context on GitHub Docs and GitHub Pages, and the team preferring Gatsby to other options.

See [#2141](https://github.com/BCDevOps/developer-experience/issues/2141) for previous proof of concept work using VuePress.

New mono-repo for documentation: https://github.com/bcgov/platform-developer-docs

**Definition of done**
- [x] Code for the Gatsby deployment and source documents are stored in a GitHub repository ([bcgov/platform-developer-docs](https://github.com/bcgov/platform-developer-docs))
- [x] An instance of the Gatsby app is deployed on `bdec76-dev`: http://platform-developer-docs-bdec76-dev.apps.silver.devops.gov.bc.ca/

**Comment by ty2k** [comment]
During the March 10-30 sprint, I did initial scaffolding work on the application and have it running locally on my machine.

![Screen Shot 2022-03-30 at 4.03.06 PM.png](https://images.zenhubusercontent.com/5ec709daacce4878764898c4/656854af-4eb0-407b-b3ca-6f2a595db845)

As next steps, I am working on a Dockerfile to deploy the application in OpenShift, and I'm preparing a pull request to the [@bcgov/platform-developer-docs repository](https://github.com/bcgov/platform-developer-docs) where the application will live alongside the documentation files. This way, pull requests that update the documentation can be used to re-build and deploy the application.

**Comment by ty2k** [comment]
During the April 21 - May 11 sprint, I committed my Gatsby application code to the [documentation repository](https://github.com/bcgov/platform-developer-docs), moving the docs and images into the Gatsby `src` directory for Gatsby to build pages from them without any extra scaffolding.

I created a Docker deployment using the `./Dockerfile.prod` configuration, building the app using images for `node:lts-alpine` and `nginxinc/nginx-unprivileged:1.20`. This deployment is in the `bdec76-dev` namespace here: [Platform Developer Docs](http://platform-developer-docs-bdec76-dev.apps.silver.devops.gov.bc.ca/).
